plugins/csp-web-api/src/Plugin.cpp

```diff
@@ -345,73 +386,81 @@ void Plugin::update() {
     // server's worker thread that the screen shot is done.
     if (mCaptureAtFrame > 0 &&
         mCaptureAtFrame == GetVistaSystem()->GetFrameLoop()->GetFrameCount()) {
-      logger().debug("Capturing capture for /capture request: resolution = {}x{}, show gui = {}",
-          mCaptureWidth, mCaptureHeight, mCaptureGui);
+      logger().debug("Capturing capture for /capture request: resolution = {}x{}, show gui = {}, "
+                     "depth = {}, format = {}",
+          mCaptureWidth, mCaptureHeight, mCaptureGui, mCaptureDepth, mCaptureFormat);
 
       auto* window = GetVistaSystem()->GetDisplayManager()->GetWindows().begin()->second;
       window->GetWindowProperties()->GetSize(mCaptureWidth, mCaptureHeight);
 
+      // We encode image data in the main thread as this is not thread-safe.
+      stbi_flip_vertically_on_write(1);
```

Comment:
Is it possible to cache the value of this flag? So we can reset it to the previous value, even if it wasn't 0?

Reply:
Well, there is no "official" getter. I think we could just access the global `static int stbi__flip_vertically_on_write` but this is not intended, I guess. And it may be even impossible with MSVC as it's not exported. So we should simply reset it to its default state. Or do you see a better way?

Reply:
Yeah, I looked at the definition of the variable. As a **global** `static` in a header file, we don't even need to bother to reset it at all, since each `.cpp` file has it's own version. So we can leave it as is.